redisson/src/main/java/org/redisson/RedissonSpinLock.java

```diff
@@ -0,0 +1,482 @@
+/**
+ * Copyright (c) 2013-2020 Nikita Koksharov
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.redisson;
+
+import io.netty.util.concurrent.GlobalEventExecutor;
+import org.redisson.api.RFuture;
+import org.redisson.client.RedisException;
+import org.redisson.client.codec.LongCodec;
+import org.redisson.client.protocol.RedisCommands;
+import org.redisson.client.protocol.RedisStrictCommand;
+import org.redisson.command.CommandAsyncExecutor;
+import org.redisson.misc.RPromise;
+import org.redisson.misc.RedissonPromise;
+
+import java.security.SecureRandom;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Distributed implementation of {@link java.util.concurrent.locks.Lock}
+ * Implements reentrant lock.<br>
+ * Lock will be removed automatically if client disconnects.
+ * <p>
+ * Implements a <b>non-fair</b> locking so doesn't guarantees an acquire order.
+ *
+ * @author Danila Varatyntsev
+ */
+public class RedissonSpinLock extends RedissonBaseLock {
+
+    public interface BackOffOptions {
+        BackOffPolicy create();
+    }
+
+    public interface BackOffPolicy {
+
+        long getNextSleepPeriod();
+    }
+
+    public static class ExponentialBackOffOptions implements BackOffOptions {
+        private long maxDelay = 128;
+        private long initialDelay = 1;
+        private int multiplier = 2;
+
+        @Override
+        public BackOffPolicy create() {
+            return new ExponentialBackOffPolicy(initialDelay, maxDelay, multiplier);
+        }
+
+        public long getMaxDelay() {
+            return maxDelay;
+        }
+
+        public ExponentialBackOffOptions maxDelay(long maxDelay) {
+            this.maxDelay = maxDelay;
+            return this;
+        }
+
+        public long getInitialDelay() {
+            return initialDelay;
+        }
+
+        public ExponentialBackOffOptions initialDelay(long initialDelay) {
+            this.initialDelay = initialDelay;
+            return this;
+        }
+
+        public int getMultiplier() {
+            return multiplier;
+        }
+
+        public ExponentialBackOffOptions multiplier(int multiplier) {
+            this.multiplier = multiplier;
+            return this;
+        }
+    }
+
+    private static final class ExponentialBackOffPolicy implements BackOffPolicy {
+
+        private static final SecureRandom RANDOM = new SecureRandom();
```

Comment:
I don't think we do need to use SecureRandom here. ThreadLocalRandom.current() should be enough